client/my-sites/plan-features/index.jsx

```diff
@@ -139,6 +140,69 @@ class PlanFeatures extends Component {
 		);
 	}
 
+	hasDiscountNotice() {
+		const { canPurchase, hasPlaceholders, withDiscount } = this.props;
+		const bannerContainer = this.getBannerContainer();
+		if ( ! bannerContainer ) {
+			return false;
+		}
+
+		const activeDiscount = getDiscountByName( withDiscount );
+		if ( ! activeDiscount || hasPlaceholders || ! canPurchase ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	renderCreditNotice() {
+		const {
+			canPurchase,
+			hasPlaceholders,
+			translate,
+			planCredits,
+			planProperties,
+			showPlanCreditsApplied,
+		} = this.props;
+		const bannerContainer = this.getBannerContainer();
+		if (
+			hasPlaceholders ||
+			! canPurchase ||
+			! bannerContainer ||
+			! showPlanCreditsApplied ||
+			! planCredits
+		) {
+			return null;
+		}
+
+		return ReactDOM.createPortal(
+			<Notice
+				className="plan-features__notice-credits"
+				showDismiss={ false }
+				icon="info-outline"
+				status="is-success"
+			>
+				{ translate(
+					'You have {{b}}%(credits)s{{/b}} in upgrade credits available! ' +
+						'Apply the value of your current plan towards an upgrade before your credits expire!',
```

Comment:
I think I would get rid of the double exclamation marks here, a little too intense. Keep the first one and change the second to just a period, I'd say.